migration/main/README.md

```diff
@@ -0,0 +1,128 @@
+---
+from: v0.43
+to: main
+---
+
+# Main upgrade and migration
+
+<!-- begin preamble --->
+
+> [!caution]
+> This is the upgrade and migration process for the next version of Welkin Apps not intended for production!
+>
+> Upgrade is supported from any patch version of the major or minor version stated in the `from` field above!
+
+> [!tip]
+> Contributors please see the [migration readme](../README.md) for instructions on how to write migration steps.
+
+<!--- end preamble --->
+
+## Prerequisites
+
+- [ ] Read through [the changelog of the new version](../../changelog) for notices.
+    - This includes platform administrator, application developer, and security notices as they will detail any breaking changes.
+- [ ] Notify users of the environment before the upgrade starts.
+- [ ] Ensure the environment has no pending changes.
+- [ ] Ensure the environment is in a good state, including the status of backups and other jobs:
+
+    ```bash
+    ./bin/ck8s test sc|wc
+    ./bin/ck8s ops kubectl sc|wc get pods -A -o custom-columns=NAMESPACE:metadata.namespace,POD:metadata.name,READY-false:status.containerStatuses[*].ready,REASON:status.containerStatuses[*].state.terminated.reason | grep false | grep -v Completed
+    ./bin/ck8s ops kubectl sc|wc get jobs -A
+    ./bin/ck8s ops kubectl sc|wc get nodes
+    ./bin/ck8s ops helm sc|wc list -A --all
+    velero get backup
+    ```
+
+- [ ] Disable alert notifications during the upgrade. e.g using [Alertmanager silences](https://prometheus.io/docs/alerting/latest/alertmanager/#silences).
+
+## Automatic method
+
+> [!tip]
+> It is possible to either upgrade both the service and workload cluster together, or upgrade service clusters and workload clusters separately by supplying the upgrade command with the `both`, `sc`, or `wc` target cluster arguments.
+
+1. Switch to the main branch and pull the latest changes:
+
+    ```bash
+    git switch main
+    git pull
+    ```
+
+1. Prepare upgrade
+
+    > _Non-disruptive, done before maintenance window._
+
+    ```bash
+    ./bin/ck8s upgrade <target-cluster> main prepare
+
+    # Check if any NetworkPolicy IPs need to be updated
+    ./bin/ck8s update-ips <target-cluster> dry-run
+    # Apply the changes if you agree with them
+    ./bin/ck8s update-ips <target-cluster> apply
```

Comment:
Personally I would prefer to use an environment variable here for specifying the cluster or clusters to apply the upgrade to, to make it more "copy-paste-able". So maybe have a separate step for setting a variable, e.g `CK8S_CLUSTER` which we are using in the manual steps:
```suggestion
    ./bin/ck8s upgrade ${CK8S_CLUSTER} main prepare

    # Check if any NetworkPolicy IPs need to be updated
    ./bin/ck8s update-ips ${CK8S_CLUSTER} dry-run
    # Apply the changes if you agree with them
    ./bin/ck8s update-ips ${CK8S_CLUSTER} apply
```

Reply:
I've made targeting a separate step, so once that is done the rest is copypasteable.